File: lib/feedback-service.ts
/**
 * Feedback service for handling user feedback functionality
 * Provides methods to open feedback forms and handle feedback events
 */

import type { GiveFeedbackMessage, FeedbackData } from '@/lib/types/unity-bridge';

const FEEDBACK_URL = 'https://docs.google.com/forms/d/e/1FAIpQLScLNB8uP_wrdDLBJ2J2E28BtxbK1oYyfBOWPu-ubOcuqgVvPQ/viewform?usp=header';

/**
 * Opens the feedback form in a new tab
 * @param data Optional feedback data for tracking/analytics
 */
export function giveFeedback(data: FeedbackData = {}) {
  console.log('[FeedbackService] Give feedback called with data:', data);
  
  // Open feedback URL in new tab
  window.open(FEEDBACK_URL, '_blank', 'noopener,noreferrer');
}

/**
 * Handles feedback messages from Unity games
 * @param message The feedback message from Unity
 */
export function handleFeedbackMessage(message: GiveFeedbackMessage) {
  console.log('[FeedbackService] Handling Unity feedback message:', message);
  
  // Call giveFeedback with the message data
  giveFeedback(message.data || {});
}

/**
 * @deprecated Use handleFeedbackMessage instead
 * Legacy function for backward compatibility
 */
export function handleFeedbackEvent(eventData: any) {
  console.log('[FeedbackService] Handling legacy Unity feedback event:', eventData);
  
  // Convert legacy event data to new message format if needed
  const data = eventData?.data || {};
  giveFeedback(data);
} 

File: game_skeletons/SubwaySurfers/Assets/ReusablePatterns/SharedCore/Scripts/Runtime/FeedbackSystem/FeedbackService.cs
using UnityEngine;
using AIEduChatbot.UnityReactBridge.Core;
using AIEduChatbot.UnityReactBridge.Data;
namespace AIEduChatbot.SharedCore
{
    /// <summary>
    /// Service for handling feedback functionality across the game
    /// Provides methods to give feedback either through React bridge or direct URL opening
    /// </summary>
    public static class FeedbackService
    {
        private const string FEEDBACK_URL = "https://docs.google.com/forms/d/e/1FAIpQLScLNB8uP_wrdDLBJ2J2E28BtxbK1oYyfBOWPu-ubOcuqgVvPQ/viewform?usp=header";

        /// <summary>
        /// Opens the feedback form. Uses React bridge if available, otherwise opens URL directly.
        /// </summary>
        /// <param name="data">Additional feedback data to send (optional)</param>
        public static void GiveFeedback(FeedbackData data = null)
        {
            Debug.Log("[FeedbackService] GiveFeedback called");

            // Check if React bridge is available
            if (ReactBridge.IsAvailable)
            {
                Debug.Log("[FeedbackService] React bridge available, sending GiveFeedbackMessage");

                // Send message through React bridge
                var feedbackMessage = new GiveFeedbackMessage(data);
                ReactBridge.SendGameMessage(feedbackMessage);
            }
            else
            {
                Debug.Log("[FeedbackService] React bridge not available, opening URL directly");

                // Fallback to direct URL opening
                Application.OpenURL(FEEDBACK_URL);
            }
        }

        /// <summary>
        /// Opens the feedback form with default Unity source data
        /// </summary>
        public static void GiveFeedback()
        {
            GiveFeedback(new FeedbackData
            {
                Source = "unity",
                Metadata = new { origin = "game_button" }
            });
        }
    }
}